salt/kubernetes-minion/kubelet.jinja

```diff
@@ -12,8 +12,8 @@ KUBELET_ADDRESS="--address=0.0.0.0"
 # The port for the info server to serve on
 KUBELET_PORT="--port=10250"
 
-# You may leave this blank to use the actual hostname
-KUBELET_HOSTNAME=""
+# Use machine ID as hostname to ensure uniqueness
+KUBELET_HOSTNAME="--hostname-override={{ grains['id'] }}"
```

Comment:
The /usr/lib/systemd/system/kubelet.service unit is simply concatenating all the variables from /etc/kubernetes/kubelet into parameters for hyperkube. 
E.g.  KUBELET_PORT="--port=10250"